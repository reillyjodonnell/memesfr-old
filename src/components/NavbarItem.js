import React, { useState } from 'react';

export default function NavbarItem(props) {
  const [openMenu, setOpenMenu] = useState(false);

  function toggleMenu() {
    setOpenMenu((prevState) => !prevState);
  }
  return (
    <>
      <div data-testid="section-1" onClick={toggleMenu}>
        {props.icon}
      </div>
      {openMenu && props.children}
    </>
  );
}
