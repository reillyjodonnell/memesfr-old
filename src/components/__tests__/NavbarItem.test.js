import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import NavbarItem from '../NavbarItem';

afterEach(() => {
  cleanup();
});

test('should render the NavbarItem component', () => {
  render(<NavbarItem />);
  const sectionHeaderElement = screen.getByTestId('section-1');
  expect(sectionHeaderElement).toBeInTheDocument();
});

test('should render the props in NavbarItem component', () => {
  const icon = 'ğŸ”¥';
  render(<NavbarItem icon={icon} />);
  const sectionHeaderElement = screen.getByTestId('section-1');
  expect(sectionHeaderElement).toHaveTextContent(icon);
});

test('should match snapshot', () => {
  const icon = 'ğŸ”¥';
  const tree = renderer.create(<NavbarItem icon={icon} />).toJSON();
  expect(tree).toMatchSnapshot();
});
