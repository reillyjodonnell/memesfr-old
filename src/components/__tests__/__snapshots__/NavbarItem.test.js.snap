// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  data-testid="section-1"
  onClick={[Function]}
>
  ðŸ”¥
</div>
`;
